dot/rpc/modules/state.go

```diff
@@ -226,6 +228,9 @@ func (sm *StateModule) GetKeysPaged(_ *http.Request, req *StateStorageKeyRequest
 		return err
 	}
 	keys, err := sm.storageAPI.GetKeysWithPrefix(req.Block, hPrefix)
+	if err != nil {
+		return err
```

Comment:
A bit out of scope, but we should return a context wrapping the error

```suggestion
		return fmt.Errorf("cannot get keys with prefix %s: %w", hPrefix, err)
```